File: Leetcode/Trees/level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
// https://leetcode.com/problems/binary-tree-level-order-traversal

class Solution {
public:
    vector<vector<int>> ans;    
    
    void findSol(TreeNode* temp, int track){
        
        if(temp){
            
            if(!track)
                ans.push_back({temp->val});
            else{
                if(track==ans.size()){
                    ans.push_back({temp->val});
                }
                else{
                    ans[track].push_back(temp->val);
                }
            }                        
                
            findSol(temp->left, track+1);
            findSol(temp->right, track+1);
        }
        
    }
    
    vector<vector<int>> levelOrder(TreeNode* root) {
        
        findSol(root, 0);
        
        return ans;
        
    }
};

File: Leetcode/Trees/level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
# https://leetcode.com/problems/binary-tree-level-order-traversal

class Solution:
    
    sol = []
    
    def traverse(self, root, track):
        
        if root is not None:        
            if track == 0:
                self.sol.append([root.val])
            else:
                if len(self.sol) == track :
                    self.sol.append([root.val])
                else:
                    self.sol[track].append(root.val)                


            self.traverse(root.left, track+1)
            self.traverse(root.right, track+1)
        
    
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        
        self.sol = []
        self.traverse(root, 0)        
        return self.sol
